x/concentrated-liquidity/fees.go

```diff
@@ -0,0 +1,183 @@
+package concentrated_liquidity
+
+import (
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils/accum"
+)
+
+const (
+	feeAccumPrefix        = "fee"
+	feeAccumNameSeparator = "/"
+	uintBase              = 10
+)
+
+var (
+	zero = sdk.ZeroDec()
+)
+
+func (k Keeper) createFeeAccumulator(ctx sdk.Context, poolId uint64) error {
+	err := accum.MakeAccumulator(ctx.KVStore(k.storeKey), getFeeAccumulatorName(poolId))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k Keeper) getFeeAccumulator(ctx sdk.Context, poolId uint64) (accum.AccumulatorObject, error) {
+	acc, err := accum.GetAccumulator(ctx.KVStore(k.storeKey), getFeeAccumulatorName(poolId))
+	if err != nil {
+		return accum.AccumulatorObject{}, err
+	}
+
+	return acc, nil
+}
+
+func (k Keeper) chargeFee(ctx sdk.Context, poolId uint64, feeUpdate sdk.DecCoin) error {
+	feeAccumulator, err := k.getFeeAccumulator(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	feeAccumulator.UpdateAccumulator(sdk.NewDecCoins(feeUpdate))
+
+	return nil
+}
+
+func (k Keeper) initializeFeeAccumulatorPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, liquidityDelta sdk.Dec) error {
+	feeAccumulator, err := k.getFeeAccumulator(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	if err := feeAccumulator.NewPositionCustom(owner.String(), zero, sdk.NewDecCoins(), nil); err != nil {
+		return err
+	}
+
```

Comment:
need to use `liquidityDelta instead of initializing them with zero values. Changed them in #